### PR TITLE
fix #44732, propagate effects of a thrown concrete call correctly

### LINKED ISSUE #44732: Dispatch failure or failure to throw error regression
Code to repro the issue:

```jl
julia> using Dash

julia> app = dash(assets_folder = "assets_compressed", compress = true);

julia> make_handler(app)
```

In 1.7:

```jl
julia> make_handler(app)
ERROR: The layout must be a component, tree of components, or a function which returns a component.
Stacktrace:
 [1] error(s::String)
   @ Base ./error.jl:33
 [2] validate_layout(layout::Nothing)
   @ Dash ~/PkgEval/Dash.jl/src/handler/make_handler.jl:44
 [3] make_handler(app::Dash.DashApp, registry::DashBase.ResourcesRegistry; check_layout::Bool)
   @ Dash ~/PkgEval/Dash.jl/src/handler/make_handler.jl:123
 [4] make_handler(app::Dash.DashApp)
   @ Dash ~/PkgEval/Dash.jl/src/handler/make_handler.jl:158
```

In 1.8:

```jl
julia> make_handler(app); # No error
```

Now, this should cleary error because what ends up getting called is

https://github.com/plotly/Dash.jl/blob/f0606e07d2479fd8b5be1da4a6a59656e24acfa3/src/handler/make_handler.jl#L123

which has these three definitions:

https://github.com/plotly/Dash.jl/blob/f0606e07d2479fd8b5be1da4a6a59656e24acfa3/src/handler/make_handler.jl#L40-L44

and `app.layout` is `nothing`:

```jl
julia> app.layout === nothing
true
```

In fact, it errors if we execute it manually:

```jl
julia> Dash.validate_layout(Dash.get_layout(app))
ERROR: The layout must be a component, tree of components, or a function which returns a component.
Stacktrace:
 [1] error(s::String)
   @ Base ./error.jl:35
 [2] validate_layout(layout::Nothing)
   @ Dash ~/PkgEval/Dash.jl/src/handler/make_handler.jl:44
```

and also when running it through the debugger:

```jl
julia> using Debugger

julia> @run make_handler(app)
ERROR: The layout must be a component, tree of components, or a function which returns a component.
````

So something here makes the dispatch go to the wrong function or the error to be swallowed.

### PR DESCRIPTION
**Pull request by aviatesk**

### CODE REVIEW

File: base/compiler/abstractinterpretation.jl [hunk 1]
```diff
@@ -743,17 +739,18 @@ function concrete_eval_call(interp::AbstractInterpreter,
     @nospecialize(f), result::MethodCallResult, arginfo::ArgInfo, sv::InferenceState)
     concrete_eval_eligible(interp, f, result, arginfo, sv) || return nothing
     args = collect_const_args(arginfo)
+    local value
     try
         value = Core._call_in_world_total(get_world_counter(interp), f, args...)
```
Comment (vtjnash) [inline base/compiler/abstractinterpretation.jl:744]:
```suggestion
    value = try
        Core._call_in_world_total(get_world_counter(interp), f, args...)
```

File: test/compiler/inline.jl [hunk 2]
```diff
@@ -1095,6 +1095,28 @@ Base.@assume_effects :consistent :effect_free :terminates_globally consteval(
...
+
+function issue44732(c::Container44732)
+    validate44732(c.x)
+    return nothing
+end
+
+let src = code_typed1(issue44732, (Container44732,))
+    @test any(isinvoke(:validate44732), src.code)
+end
+@test_throws ErrorException issue44732(Container44732(nothing))
```
Comment (vtjnash) [inline test/compiler/inline.jl:1118]:
```suggestion
@test_throws ErrorException("don't erase this error!") issue44732(Container44732(nothing))
```

**Comment by aviatesk** [comment]
@nanosoldier `runbenchmarks("inference", vs=":master")`

**Comment by nanosoldier** [comment]
[Your benchmark job](https://github.com/JuliaLang/julia/pull/44762#issuecomment-1081333884) has completed - no performance regressions were detected. A full report can be found [here](https://github.com/JuliaCI/NanosoldierReports/blob/master/benchmark/by_hash/b8b2354_vs_c919811/report.md).

**Pull request merged**